scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java

```diff
@@ -1915,20 +1918,31 @@ public void endActivity(Body body) {
 
     @Override
     @ImmediateService
-    public boolean addJobSignal(String sessionId, String jobId, String signal) {
+    public Set<String> addJobSignal(String sessionId, String jobId, String signal) throws SignalApiException {
         try {
+            publicStore.putIfAbsent(SIGNAL_ORIGINATOR, SIGNAL_TASK_ID, signalsChannel, jobId, new HashSet<String>());
```

Comment:
@fviale 
I added this line because when the user wants to send a ready signal for the first time (i.e., the set of  job signals is null), we have to initialize the set of  job signals with a HashSet<String>.

Reply:
ok